custom_components/tahoma/binary_sensor.py

```diff
@@ -1,64 +1,101 @@
 """Support for TaHoma binary sensors."""
-from typing import Optional
 
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Callable
+
+from homeassistant.components import binary_sensor
 from homeassistant.components.binary_sensor import (
-    DEVICE_CLASS_MOTION,
-    DEVICE_CLASS_OCCUPANCY,
-    DEVICE_CLASS_OPENING,
-    DEVICE_CLASS_SMOKE,
-    DOMAIN as BINARY_SENSOR,
     BinarySensorEntity,
+    BinarySensorEntityDescription,
 )
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DOMAIN
+from .coordinator import TahomaDataUpdateCoordinator
 from .tahoma_entity import TahomaEntity
 
-CORE_ASSEMBLY_STATE = "core:AssemblyState"
-CORE_BUTTON_STATE = "core:ButtonState"
-CORE_CONTACT_STATE = "core:ContactState"
-CORE_GAS_DETECTION_STATE = "core:GasDetectionState"
-CORE_OCCUPANCY_STATE = "core:OccupancyState"
-CORE_OPENING_STATE = "core:OpeningState"
-CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"
-CORE_RAIN_STATE = "core:RainState"
-CORE_SMOKE_STATE = "core:SmokeState"
-CORE_THREE_WAY_HANDLE_DIRECTION_STATE = "core:ThreeWayHandleDirectionState"
-CORE_VIBRATION_STATE = "core:VibrationState"
-CORE_WATER_DETECTION_STATE = "core:WaterDetectionState"
-
-DEVICE_CLASS_BUTTON = "button"
-DEVICE_CLASS_GAS = "gas"
-DEVICE_CLASS_RAIN = "rain"
-DEVICE_CLASS_WATER = "water"
-
-ICON_WATER = "mdi:water"
-ICON_WATER_OFF = "mdi:water-off"
-ICON_WAVES = "mdi:waves"
-ICON_WEATHER_RAINY = "mdi:weather-rainy"
-
-IO_VIBRATION_DETECTED_STATE = "io:VibrationDetectedState"
-
 STATE_OPEN = "open"
 STATE_PERSON_INSIDE = "personInside"
 STATE_DETECTED = "detected"
-STATE_PRESSED = "pressed"
-
-TAHOMA_BINARY_SENSOR_DEVICE_CLASSES = {
-    "AirFlowSensor": DEVICE_CLASS_GAS,
-    "CarButtonSensor": DEVICE_CLASS_BUTTON,
-    "ContactSensor": DEVICE_CLASS_OPENING,
-    "MotionSensor": DEVICE_CLASS_MOTION,
-    "OccupancySensor": DEVICE_CLASS_OCCUPANCY,
-    "RainSensor": DEVICE_CLASS_RAIN,
-    "SirenStatus": DEVICE_CLASS_OPENING,
-    "SmokeSensor": DEVICE_CLASS_SMOKE,
-    "WaterDetectionSensor": DEVICE_CLASS_WATER,
-    "WaterSensor": DEVICE_CLASS_WATER,
-    "WindowHandle": DEVICE_CLASS_OPENING,
-}
+
+
+@dataclass
+class OverkizBinarySensorDescription(BinarySensorEntityDescription):
+    """Class to describe a Overkiz binary sensor."""
+
+    is_on: Callable[[Any], Any] = lambda state: state
+
+
+BINARY_SENSOR_DESCRIPTIONS = [
+    # RainSensor/RainSensor
+    OverkizBinarySensorDescription(
```

Comment:
You removed 13 constants, but only added 9 descriptions, it’s on purpose?

Reply:
Yes, I removed `core:ThreeWayHandleDirectionState` since this is a property that exists when other supported properties are available as well. I don't want to add multiple sensors for the same devices, that output the same info.

`core:ButtonState` since it is probably not used and an edge case that doesn't belong in `binary_sensor`.

However, it seems I missed `CORE_OPENING_STATE = "core:OpeningState"` and `CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"`. Will add them soon.